src/stm32/Kconfig

```diff
@@ -244,6 +245,15 @@ config STM32_DFU_ROM_ADDRESS
     default 0
 
 
+config N32G452
+    bool "Build for Nation N32G45x MCU"
+    depends on MACH_STM32F103 && LOW_LEVEL_OPTIONS
+    default n
+    help
+        Some Voxelab Aquila motherboards have N32G452 chip instead of
+        GD32F103. This cortex-M4 chip is very similar to F103 but
+        have different ADC. Enable this option to make ADC working.
```

Comment:
I think it may be preferable to add a new MACH_N32G452 definition to the main "Processor model" menu.  If there are places in Kconfig or the code that are using CONFIG_MACH_STM32F103 where they should be using CONFIG_MACH_STM32F1 then I think we should just update those references.  That way, the MACH_N32G452 could be a MACH_STM32F1 but not a MACH_STM32F103.